### PR TITLE
11 happy path query on articles

### PR DESCRIPTION
**Pull request by Jord000**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -407,3 +407,41 @@ describe('GET /api/users', () => {
...
+      .expect(200)
+      .then(({ body:{articles}}) => {
+           articles.forEach((article) => {
+          expect(article.topic).toBe('cats');
+        });
+      });
+  });
+  test('should give back empty object with topic that doesnt exist', () => {
+    return request(app)
+      .get('/api/articles?topic=neverheardofit')
```
Comment (nickkagame) [inline __tests__/app.test.js]: don't this this should be a 200.  This topic doesn't exist! So should be 404 I think

File: __tests__/app.test.js [hunk 2]
```diff
@@ -407,3 +407,41 @@ describe('GET /api/users', () => {
...
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toEqual([]);
+      });
+  });
+  test('should handle error with category that doesnt exist', () => {
+    return request(app)
+      .get('/api/articles?thisisanerror')
```
Comment (nickkagame) [inline __tests__/app.test.js]: This is not a category that doesn't exist, it's a query label, and should probably not be tested for.  It's just gonna be what is attached as the key to you req.query ({thisisanerror: 'blah blah}).  So I'd remove this test.

File: __tests__/app.test.js (hunk 1 above)
Comment (nickkagame) [reply __tests__/app.test.js]: Should only be an empty array for a topic that exists but doesn't yet have any articles

**Review by nickkagame — CHANGES_REQUESTED** [review]
Hi jordan, nice work, but I think there's a bit of confusion. This task is just to be able to filter by topic.   So the util to check that the query tag is correct is a bit unnessecary at this stage.  Also the error behaviours to check for are - Status 200, accepts topic query, e.g. ?topic=coding.
should check that all returned objects have the queried topic.
Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.
Status 404, non-existent topic query, e.g. ?topic=bananas.

You'll have to check that a topic exists in the DB rather than check the query key against a green list.   

I hope this makes sense!

File: __tests__/app.test.js (hunk 2 above)
Comment (Jord000) [reply __tests__/app.test.js]: thanks Nick I've updated all of these and it should be better now. My only issue is on a query with an invalid query input. Not sure if this should be error handled or if this is just a miss-use of the API and its ok that the default articles are sent back

**Review by nickkagame — APPROVED** [review]
Great work jordan!

**Pull request merged**